eth/downloader/downloader.go

```diff
@@ -440,10 +440,14 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.I
 	}
 	height := latest.Number.Uint64()
 
-	origin, err := d.findAncestor(p, latest)
-	if err != nil {
-		return err
+	var origin = uint64(0)
+	if d.blockchain.CurrentHeader().Number.Uint64() > 0 {
+		origin, err = d.findAncestor(p, latest)
```

Comment:
Only negotiate a non-zero common ancestor if our local head is actually above genesis.